src/components/otp-input/otp-input.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useRef } from 'react';
+import { HStack, Input } from 'native-base';
+
+interface OTPInputProps {
+  length: number;
+  otp: string[];
+  setOtp: (otp: string[]) => void;
+}
+
+const OTPInput: React.FC<OTPInputProps> = ({ length, otp, setOtp }) => {
+  const inputsRef = useRef<Array<any>>([]);
+
+  const handleChange = (text: string, index: number) => {
+    const newOtp = [...otp];
+    newOtp[index] = text;
+    setOtp(newOtp);
+
+    if (text && index < length - 1) {
+      inputsRef.current[index + 1].focus();
+    }
+  };
+
+  const handleKeyPress = (e: any, index: number) => {
+    if (e.nativeEvent.key === 'Backspace' && index > 0 && otp[index] === '') {
+      inputsRef.current[index - 1].focus();
+    }
+  };
+
+  return (
+    <HStack space={2}>
+      {Array(length)
+        .fill('')
+        .map((_, index) => (
+          <Input
+            ref={el => (inputsRef.current[index] = el)}
+            value={otp[index]}
+            onChangeText={text => handleChange(text, index)}
+            onKeyPress={e => handleKeyPress(e, index)}
+            keyboardType="numeric"
+            maxLength={1}
+            width={10}
+            height={10}
+            textAlign="center"
+            variant="unstyled"
+            borderBottomWidth={2}
+            borderRadius={0}
+            _focus={{
+              borderColor: 'primary',
+            }}
```

Comment:
I would expect input component defined once and don't expect the styles to be overriden in each page.

Can we define input component of type primary, secondary as well other standard abstractions? (we can use [baseweb input abstraction as a reference](https://baseweb.design/components/input/))